src/imap/idle.rs

```diff
@@ -41,6 +42,10 @@ impl Session {
             bail!("IMAP IDLE protocol failed to init/complete: {}", err);
         }
 
+        // At this point IDLE command was sent and we received a "+ idling" response.
+        // Reset the read timeout on the socket and give control back to async-imap.
```

Comment:
What does "give control back to async-imap." mean?

This should explain why we are doing this and why only the read timeout, not the write timeout, e.g. (assuming that's correct?):

```
At this point IDLE command was sent and we received a "+ idling" response.
We will now read from the TCP stream without getting any data for up to
almost half an hour. If we didn't disable the read timeout, we would
get a timeout after `IMAP_TIMEOUT`, which is a lot shorter than  half an hour.
```